### PR TITLE
Car Port: Toyota Avalon Hybrid 2019

### PR DESCRIPTION
**Pull request by jbates9011**

### CODE REVIEW

File: selfdrive/car/toyota/values.py [hunk 1]
```diff
@@ -204,6 +205,9 @@ class CAR:
...
   }],
+  CAR.AVALONH: [{
+    36: 8, 37: 8, 166: 8, 170: 8, 180: 8, 295: 8, 296: 8, 426: 6, 452: 8, 466: 8, 467: 8, 550: 8, 552: 4, 560: 7, 562: 6, 581: 5, 608: 8, 610: 8, 643: 7, 658: 8, 713: 8, 728: 8, 740: 5, 761: 8, 764: 8, 800: 8, 810: 2, 812: 8, 814: 8, 818: 8, 822: 8, 824: 8, 829: 2, 830: 7, 835: 8, 836: 8, 869: 7, 870: 7, 871: 2, 888: 8, 889: 8, 891: 8, 898: 8, 900: 6, 902: 6, 905: 8, 913: 8, 918: 8, 921: 8, 933: 8, 934: 8, 935: 8, 944: 8, 945: 8, 950: 8, 951: 8, 953: 8, 955: 8, 956: 8, 971: 7, 975: 5, 983: 8, 984: 8, 993: 8, 998: 5, 999: 7, 1000: 8, 1001: 8, 1002: 8, 1011: 8, 1014: 8, 1017: 8, 1020: 8, 1041: 8, 1042: 8, 1044: 8, 1056: 8, 1057: 8, 1059: 1, 1071: 8, 1076: 8, 1077: 8, 1082: 8, 1084: 8, 1085: 8, 1086: 8, 1114: 8, 1132: 8, 1161: 8, 1162: 8, 1163: 8, 1164: 8, 1165: 8, 1166: 8, 1167: 8, 1227: 8, 1228: 8, 1235: 8, 1237: 8, 1263: 8, 1264: 8, 1279: 8, 1412: 8, 1541: 8, 1552: 8, 1553: 8, 1556: 8, 1557: 8, 1568: 8, 1570: 8, 1571: 8, 1572: 8, 1595: 8, 1745: 8, 1779: 8, 1786: 8, 1787: 8, 1788: 8, 1789: 8
+  }],
```
Comment (ShaneSmiskol) [inline selfdrive/car/toyota/values.py:210]: Remove the CAN fingerprint entry, it will be deprecated soon.

File: selfdrive/car/toyota/values.py [hunk 2]
```diff
@@ -36,6 +36,7 @@ class CAR:
   HIGHLANDERH = "TOYOTA HIGHLANDER HYBRID 2018"
   HIGHLANDERH_TSS2 = "TOYOTA HIGHLANDER HYBRID 2020"
   AVALON = "TOYOTA AVALON 2016"
+  AVALONH = "TOYOTA AVALON HYBRID 2019"
```
Comment (ShaneSmiskol) [inline selfdrive/car/toyota/values.py:39]: Rename the variable to `AVALONH_2019`

Comment (ShaneSmiskol) [inline selfdrive/car/toyota/values.py:39]:
```suggestion
  AVALONH_2019 = "TOYOTA AVALON HYBRID 2019"
```

File: selfdrive/car/toyota/values.py [hunk 3]
```diff
@@ -300,6 +304,26 @@ class CAR:
       b'8646F0703000\x00\x00\x00\x00',
     ],
   },
+  CAR.AVALONH: {
```
Comment (ShaneSmiskol) [inline selfdrive/car/toyota/values.py:307]:
```suggestion
  CAR.AVALONH_2019: {
```

File: selfdrive/car/toyota/values.py [hunk 4]
```diff
@@ -1639,6 +1663,7 @@ class CAR:
   CAR.HIGHLANDERH: dbc_dict('toyota_highlander_hybrid_2018_pt_generated', 'toyota_adas'),
   CAR.HIGHLANDERH_TSS2: dbc_dict('toyota_nodsu_hybrid_pt_generated', 'toyota_tss2_adas'),
   CAR.AVALON: dbc_dict('toyota_avalon_2017_pt_generated', 'toyota_adas'),
+  CAR.AVALONH: dbc_dict('toyota_nodsu_hybrid_pt_generated', 'toyota_adas'),
```
Comment (ShaneSmiskol) [inline selfdrive/car/toyota/values.py:1666]:
```suggestion
  CAR.AVALONH_2019: dbc_dict('toyota_nodsu_hybrid_pt_generated', 'toyota_adas'),
```

File: selfdrive/car/toyota/values.py (hunk 3 above)
Comment (jbates9011) [reply selfdrive/car/toyota/values.py]: Agree.

File: selfdrive/car/toyota/values.py (hunk 1 above)
Comment (jbates9011) [reply selfdrive/car/toyota/values.py]: Agree.

**Comment by adeebshihadeh** [comment]
Looks good, except we'll need a test route added [here](https://github.com/commaai/openpilot/blob/master/selfdrive/test/test_routes.py).

**Comment by jbates9011** [comment]
Thanks Adeeb.  I appreciate your assistance in getting the PR merged!

 

Jim Bates

**Pull request merged**